apps/src/javalab/JavalabEditor.jsx

```diff
@@ -61,16 +64,38 @@ class JavalabEditor extends React.Component {
   }
 
   componentDidMount() {
+    const {isDarkMode} = this.props;
+    const extensions = [...editorSetup];
+
+    if (isDarkMode) {
+      extensions.push(tagExtension('style', oneDark));
+    } else {
+      extensions.push(tagExtension('style', lightMode));
+    }
     this.editor = new EditorView({
       state: EditorState.create({
         doc: this.props.editorText,
-        extensions: editorSetup
+        extensions: extensions
       }),
       parent: this._codeMirror,
       dispatch: this.dispatchEditorChange()
     });
   }
 
+  componentDidUpdate(prevProps) {
+    if (prevProps.isDarkMode !== this.props.isDarkMode) {
```

Comment:
This is the "mainline" scenario for changing between dark and light mode. Adding a unit test here will ensure this change persists well into the future.